src/core/utils/spec-preprocess.ts

```diff
@@ -187,6 +192,11 @@ export function traverseToFixSpecDownstream(spec: GoslingSpec | SingleView, pare
 
         const linkID = uuid.v4();
         tracks.forEach((track, i, array) => {
+            // ID should be assigned to each view and track for an API usage
+            if (!track.id) {
+                track.id = uuid.v4();
+            }
+
```

Comment:
Okay great, glad this isn't happening in HiGlass anymore!

Reply:
It is. I think these lines can be removed.